include/nlohmann/detail/iterators/iter_impl.hpp

```diff
@@ -463,11 +463,17 @@ class iter_impl // NOLINT(cppcoreguidelines-special-member-functions,hicpp-speci
 
     /*!
     @brief comparison: equal
-    @pre The iterator is initialized; i.e. `m_object != nullptr`.
+    @pre (1) The iterator is initialized; i.e. `m_object != nullptr`, or (2) both iterators are value-initialized.
```

Comment:
Should this say `(1) Both iterators are initialized to point to the same object`